blueocean-dashboard/src/main/js/creation/flow2/FlowManager.js

```diff
@@ -164,6 +164,18 @@ export default class FlowManager {
         this.stateId = stateId;
 
         LOGGER.debug(`changed stateId to ${stateId}`);
+
+        if ('STEP_COMPLETE_MISSING_JENKINSFILE' === this.stateId) {
+            setTimeout(() => {
+                this._navigateToPipelineEditor();
+                ToastService.newToast({ text: 'No Jenkinsfiles were found in this repository. Get started by creating one.' });
+            }, this.redirectTimeout);
+        }
+    }
+
+    _navigateToPipelineEditor() {
+        const url = `/organizations/${AppConfig.getOrganizationName()}/pipeline-editor/${encodeURIComponent(this.pipelineName)}/`;
+        this.completeFlow({ url })
```

Comment:
`FlowManager` is meant to be a fairly generic abstraction.. I'm not sure this is the right place for this code.

Reply:
All flows should consistently drop to the editor if it's supported (ideally, that reported status).

Reply:
`FlowManager` is used in Thor's project for flows that have nothing to do with creation.